File: slf4j-messager/src/main/java/io/openapiprocessor/slf4j/messager/MessagerLoggerFactory.java
/*
 * Copyright 2022 https://github.com/openapi-processor/openapi-processor-base
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.slf4j.messager;

import org.slf4j.ILoggerFactory;
import org.slf4j.Logger;

public class MessagerLoggerFactory implements ILoggerFactory {

    public MessagerLoggerFactory() {
        MessagerLogger.init ();
    }

    @Override
    public Logger getLogger (String name) {
        return new MessagerLogger (name, new CurrentMessager ());
    }
}

File: openapi-processor-core/src/main/kotlin/io/openapiprocessor/core/logging/MessagerLoggerFactory.kt
/*
 * Copyright 2022 https://github.com/openapi-processor/openapi-processor-core
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.core.logging

import javax.annotation.processing.Messager

class MessagerLoggerFactory(private val messager: Messager): LoggerFactory {
    override fun getLogger(name: String): Logger {
        return MessagerLogger(messager, name)
    }
}
